core/package/index.js

```diff
@@ -181,7 +193,27 @@ class Package {
    * @returns {Promise} resolves when write finished
    */
   serialize() {
-    return writePkg(this.manifestLocation, this[PKG]).then(() => this);
+    return writePkg(this.manifestLocation, this[PKG])
+      .then(() => this.writePkgLock())
+      .then(() => this);
+  }
+
+  /**
+   * Write package-lock changes to disk
+   * @returns {Promise} resolves when write finished
+   */
+  writePkgLock() {
+    // We only care about version in the lock file; a package's name never changes
+    const lockChanges = { version: this[PKG].version };
+
+    const tasks = [
+      () => loadJsonFile(this.lockFileLocation),
+      lockFileData => Promise.resolve(Object.assign({}, lockFileData, lockChanges)),
+      updatedLockFileData =>
+        writeJsonFile(this.lockFileLocation, updatedLockFileData, { detectIndent: true }),
+    ];
+
+    return hasPackageLock(this.lockFileLocation) ? pWaterfall(tasks) : Promise.resolve();
```

Comment:
I don't think we need `p-waterfall` just to chain promises. This check should be done way earlier, too.